lib/api/v3/utilities/custom_field_injector.rb

```diff
@@ -0,0 +1,311 @@
+        def property_name(id)
+          "customField#{id}".to_sym
+        end
+
+        def inject_version_schema(custom_field, wp_schema)
+          raise ArgumentError unless wp_schema
+
+          @class.schema_with_allowed_collection property_name(custom_field.id),
+                                                type: 'Version',
+                                                title: custom_field.name,
+                                                values_callback: -> (*) {
+                                                  wp_schema.assignable_versions
+                                                },
+                                                value_representer: Versions::VersionRepresenter,
+                                                link_factory: -> (version) {
+                                                  {
+                                                    href: api_v3_paths.version(version.id),
+                                                    title: version.name
+                                                  }
+                                                },
+                                                required: custom_field.is_required
+        end
+
+        def inject_user_schema(custom_field, wp_schema)
+          raise ArgumentError unless wp_schema
+
+          @class.schema_with_allowed_link property_name(custom_field.id),
+                                          type: 'User',
+                                          title: custom_field.name,
+                                          required: custom_field.is_required,
+                                          href_callback: -> (*) {
+                                            api_v3_paths.available_assignees(wp_schema.project.id)
+                                          }
+        end
+
+        def inject_list_schema(custom_field)
+          representer = StringObjects::StringObjectRepresenter
+          @class.schema_with_allowed_collection property_name(custom_field.id),
+                                                type: 'StringObject',
+                                                title: custom_field.name,
+                                                values_callback: -> (*) {
+                                                  custom_field.possible_values
+                                                },
+                                                value_representer: representer,
+                                                link_factory: -> (value) {
+                                                  {
+                                                    href: api_v3_paths.string_object(value),
+                                                    title: value
+                                                  }
+                                                },
+                                                required: custom_field.is_required
+        end
+
+        def inject_basic_schema(custom_field)
```

Comment:
Assignment Branch Condition size for inject_basic_schema is too high. [16.28/15]